app/screens/server/index.tsx

```diff
@@ -133,31 +136,36 @@ const Server: NavigationFunctionComponent = ({extra, launchType, launchError, th
         };
 
         const serverUrl = await getServerUrlAfterRedirect(pingUrl, !retryWithHttp);
+        NetworkManager.createClient(serverUrl);
 
         const result = await doPing(serverUrl);
 
         if (canceled) {
             return;
         }
 
-        if (result.error && retryWithHttp) {
-            const nurl = serverUrl.replace('https:', 'http:');
-            pingServer(nurl, false);
-            return;
-        } else if (result.error) {
-            setError(result.error);
-            setConnecting(false);
+        if (result.error) {
+            NetworkManager.invalidateClient[serverUrl];
+            if (retryWithHttp) {
+                const nurl = serverUrl.replace('https:', 'http:');
+                pingServer(nurl, false);
+            } else {
+                setError(result.error);
+                setConnecting(false);
+            }
+
             return;
         }
 
-        const data = await fetchConfigAndLicense();
+
+        const data = await fetchConfigAndLicense(serverUrl);
```

Comment:
perhaps we need a `fetchGenericConfigAndLicense` action that uses the generic client